qiskit/providers/aer/backends/aerbackend.py

```diff
@@ -361,7 +364,10 @@ def _get_executor(self, **run_options):
         if 'executor' in run_options:
             return run_options['executor']
         else:
-            return getattr(self._options, 'executor', None)
+            _executor = getattr(self._options, 'executor', None)
+            if _executor:
+                delattr(self._options, 'executor')
```

Comment:
I think this will have unintended consequences because the `Options` object's attributes are used to determine if a backend supports an option name. For example, in `execute()` the attributes are checked to see if a user provided kwarg is usable by the backend: https://github.com/Qiskit/qiskit-terra/blob/main/qiskit/execute_function.py#L379 

Would just doing `self._options.executor = None` do what you need here?